pyccel/ast/numpyext.py

```diff
@@ -878,27 +562,10 @@ def fill_value(self):
     def rank(self):
         return len(self.shape)
 
-    #--------------------------------------------------------------------------
-    def fprint(self, printer, lhs, stack_array=False):
-        """Fortran print."""
-
-        lhs_code = printer(lhs)
-        stmts = []
-
-        # Create statement for initialization
-        if self.fill_value is not None:
-            init_value = printer(self.fill_value)
-            code_init = '{0} = {1}'.format(lhs_code, init_value)
-            stmts.append(code_init)
-
-        if len(stmts) == 0:
-            return ''
-        else:
-            return '\n'.join(stmts) + '\n'
-
 #==============================================================================
-class NumpyEmpty(NumpyFull):
-    """ Represents a call to numpy.empty for code generation.
+class NumpyAutoFill(NumpyFull):
+    """ Abstract class for all classes which inherit from NumpyFull but
+        the fill_value is implicitly specified
```

Comment:
Why is this class needed?

Reply:
NumpyZeros, NumpyOnes was inheriting from NumpyEmpty, so in the printing NumpyEmpty could be ether NumpyZeros or NumpyOnes in every isinstance block and this not good the new implementation of printing, this class fix that conflict and prevent repeating the same `__new__` function in NumpyEmpty.

Reply:
>so in the printing NumpyEmpty could be ether NumpyZeros or NumpyOnes in every isinstance block and this not good the new

Put another way, `isinstance` couldn't be used to handle the specific problem of `NumpyEmpty` in an `Assign` as it was also catching the subclasses `NumpyZeros` and `NumpyOnes`

Reply:
Thanks for the explanation. This should be added to the PR's description.